CleverGirl/CleverGirl/Helper/AIHelper.cs

```diff
@@ -53,16 +59,39 @@ public class AIHelper {
 
             AttackParams attackParams = new AttackParams(attackType, unit, target as AbstractActor, attackPosition, weaponList.Count, useRevengeBonus);
 
-            float totalExpectedDam = 0f;
-            for (int i = 0; i < weaponList.Count; i++) {
-                CondensedWeapon cWeapon = weaponList[i];
-                totalExpectedDam += CalculateWeaponDamageEV(cWeapon, unitForBVContext.BehaviorTree, attackParams, unit, attackPosition, target, targetPosition);
+            Dictionary<bool, float> totalExpectedDam = new Dictionary<bool, float>
+            {
+                {false, 0},
+                {true, 0}
+            };
+            foreach (var cWeapon in weaponList)
+            {
+                cWeapon.ApplyAmmoMode();
+                bool isArtillery = cWeapon.First.IsArtillery();
+                totalExpectedDam[isArtillery] += CalculateWeaponDamageEV(cWeapon, unitForBVContext.BehaviorTree, attackParams, unit, attackPosition, target, targetPosition);
+                cWeapon.RestoreBaseAmmoMode();
             }
 
             float blowQualityMultiplier = unit.Combat.ToHit.GetBlowQualityMultiplier(attackParams.Quality);
-            float totalDam = totalExpectedDam * blowQualityMultiplier;
+            float artilleryDamage = totalExpectedDam.GetValueOrDefault(true, 0f) * blowQualityMultiplier;
```

Comment:
FWIW I think using blowQualityMultiplier on my end might be a carry over from HBS, that's not actually used? I vaguely recall it being used for the 'glancing hit' mechanic, but I think it's mostly a 1f at this point?

Reply:
From what I can see it actually does check brace, sensorlock etc to find a guard level of the target, which translate to blow quality.